x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/lib/mappings_validator.test.ts

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { validateMappings, validateProperties, isObject } from './mappings_validator';
+
+jest.mock('ui/index_patterns', () => ({
+  ILLEGAL_CHARACTERS: '',
+  validateIndexPattern: () => ({}),
+}));
+
+describe('Mappings configuration validator', () => {
+  it('should convert non object to empty object', () => {
+    const tests = ['abc', 123, [], null, undefined];
+
+    tests.forEach(testValue => {
+      const { value, errors } = validateMappings(testValue as any);
+      expect(isObject(value)).toBe(true);
+      expect(errors).toBe(undefined);
+    });
+  });
+
+  it('should strip out unknown configuration', () => {
+    const mappings = {
+      dynamic: true,
+      date_detection: true,
+      numeric_detection: true,
+      dynamic_date_formats: ['abc'],
+      _source: {
+        enabled: true,
+        includes: ['abc'],
+        excludes: ['abc'],
+      },
+      properties: { title: { type: 'text' } },
+      unknown: 123,
+    };
+
+    const { value, errors } = validateMappings(mappings);
+
+    const { unknown, ...expected } = mappings;
+    expect(value).toEqual(expected);
+    expect(errors).toBe(undefined);
+  });
+
+  it('should strip out invalid configuration', () => {
```

Comment:
Nit: The above description was about stripping out values and the assertions verified no errors were returned. Since we're actually expecting errors in this test, it might make sense to rephrase this as "returns errors for invalid configuration" which will make it clearer that we're testing for two very different behaviors IMO.

Reply:
Ok I will add the "returns errors for invalid configuration" part but we are still testing that wrong configurations are filtered out.